crates/bdk/src/wallet/tx_builder.rs

```diff
@@ -839,15 +844,14 @@ pub enum TxOrdering {
 
 impl TxOrdering {
     /// Sort transaction inputs and outputs by [`TxOrdering`] variant
-    pub fn sort_tx(&self, tx: &mut Transaction) {
+    pub fn sort_tx(self, tx: &mut Transaction, rng: &mut impl RngCore) {
         match self {
             TxOrdering::Untouched => {}
             TxOrdering::Shuffle => {
-                use rand::seq::SliceRandom;
-                let mut rng = rand::thread_rng();
-                tx.input.shuffle(&mut rng);
-                tx.output.shuffle(&mut rng);
+                shuffle_slice(&mut tx.input, rng);
+                shuffle_slice(&mut tx.output, rng);
             }
+            #[allow(deprecated)]
```

Comment:
While the "swiss army knife" analogy in https://github.com/bitcoindevkit/bdk/issues/534#issuecomment-1022975295 is compelling, I'm also in favor of just removing the `Bip69Lexicographic` variant now.